patches/kaggle_datasets.py

```diff
@@ -0,0 +1,24 @@
+import os
+from kaggle_web_client import KaggleWebClient
+
+_KAGGLE_TPU_NAME_ENV_VAR_NAME = 'TPU_NAME'
+
+class KaggleDatasets:
+    GET_GCS_PATH_ENDPOINT = '/requests/CopyDatasetVersionToKnownGcsBucketRequest'
+
+    # Integration types for GCS
+    AUTO_ML = 1
+    TPU = 2
+
+    def __init__(self):
+        self.web_client = KaggleWebClient()
+        self.has_tpu = os.getenv(_KAGGLE_TPU_NAME_ENV_VAR_NAME) is not None
+
+    def get_gcs_path(self, dataset_dir: str = None) -> str:
+        integration_type = self.TPU if self.has_tpu else self.AUTO_ML
+        data = {
+            'MountSlug': dataset_dir,
+            'IntegrationType': integration_type,
+        }
+        result = self.web_client.make_post_request(data, self.GET_GCS_PATH_ENDPOINT)
```

Comment:
What's the expected behaviour if dataset_dir is None (as is the default value)? If that's not a valid case, should we fail early and give the user an error message?

Reply:
Ah I see that in your test cases it defaults to None, so I'm guessing there's a default bucket or something in that case?

Reply:
The default behavior is to copy the one dataset attached to the notebook (if there is only one). If there is more than one attached dataset, then the user is required to pass a dataset_dir or the server will fail the request.